pageserver/src/bin/pageserver.rs

```diff
@@ -99,6 +101,16 @@ fn main() -> anyhow::Result<()> {
 
     let (conf, ignored) = initialize_config(&identity_file_path, &cfg_file_path, &workdir)?;
 
+    if !dev_mode {
+        if matches!(conf.http_auth_type, AuthType::Trust) {
+            bail!("Pageserver refuses to start with HTTP API authentication disabled.\n\
+                  Run with --dev to allow running without authentication.\n\
+                  This is insecure and should only be used in development environments.");
+        }
+    } else {
+        info!("Starting in dev mode - authentication security checks are disabled");
```

Comment:
This should be a `warn!` and it's not accurate to say security checks are disabled (auth might be enabled).  Just use the same message as storage controller's main.